test/kafka_ex/utils/murmur_test.exs

```diff
@@ -0,0 +1,18 @@
+defmodule KafkaEx.Utils.MurmurTest do
+  use ExUnit.Case
+
+  alias KafkaEx.Utils.Murmur
+
+  test "murmur2" do
+    assert Murmur.murmur2("rule") == -1673595344
```

Comment:
yep, I'm seeing different results for every assert to negative numbers, but for `redundancy` or `guest` the expected value is the correct but the number are positives, also why are you using murmur2 instead of murmur3?

Reply:
Expected values are calculated with [java implementation](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L381-L419) and confirmed with [C implementation](https://github.com/aappleby/smhasher/blob/master/src/MurmurHash2.cpp).

You are getting a different number because the JS implementation returns result casted to uint32 (what umurmur2 is doing).

I chose `murmur2` to align with Java implementation. If the same hashing algorithm will be used, it is ensured that messages with the same keys are routed to the same partition regardless of the client.

I'm going to add `umurmur2` tests as well.